extract.py

```diff
@@ -8,19 +8,24 @@ def function_to_test():
 
 
 def fetch_api_plant_data(plant_id: int) -> dict:
-    """Gets plant data for a plant with given id, 
+    """Gets plant data for a plant with given id,
     returning the data as a dictionary."""
-    return
+    response = requests.get(
+        f"https://data-eng-plants-api.herokuapp.com/plants/{str(plant_id)}", timeout=100)
+    return response.json()
 
 
 def get_all_plant_data() -> list[dict]:
     """Returns all plant data for all 50 plants as
     a list of dictionaries."""
-    return
+    data_list = []
+    for i in range(0, 51):
+        plant_data = fetch_api_plant_data(i)
+        if "error" in plant_data.keys():
+            continue
+        data_list.append(plant_data)
+    return data_list
 
```

Comment:
Could we use list comprehension here instead?

Reply:
i agree!